packages/react-components/src/text/src/Text.tsx

```diff
@@ -61,9 +61,9 @@ export function InnerText(props: TextProps): ReactElement {
     );
 }
 
-export const Text = forwardRef<any, PropsWithoutForwardedRef<TextProps>>((props, ref) => (
+export const Text = slot("text", forwardRef<any, PropsWithoutForwardedRef<TextProps>>((props, ref) => (
```

Comment:
Is there a way to simplify all this? Not a big fan of `PropsWithoutForwardedRef` and `InnerPropsToProps`.

Would it be possible to do something like:

```
export interface TextProps {
    /**
     * A text can vary in size.
     */
    size?: "2xs" | "xs" | "sm" | "md" | "lg" | "xl" | "2xl" | "3xl" | "4xl" | "5xl" | "inherit";
    /**
     * An HTML element type or a custom React element type to render as.
     */
    as?: ElementType;
    /**
     * [Slot](?path=/docs/getting-started-slots--page) to render into.
     */
    slot?: string;
    /**
     * React children
     */
    children?: ReactNode;
}

export interface InnerTextProps extends TextProps {
    /**
    * @ignore
    */
    forwardedRef: ForwardedRef<any>;
}

.....

export function InnerText(props: InnerTextProps): ReactElement {
}

.....

export const Text = slot("text", forwardRef<any, TextProps>((props, ref) => (
    <InnerText {...props} forwardedRef={ref} />
)));
```

Could even be:

```
export interface InnerTextProps {
    /**
     * A text can vary in size.
     */
    size?: "2xs" | "xs" | "sm" | "md" | "lg" | "xl" | "2xl" | "3xl" | "4xl" | "5xl" | "inherit";
    /**
     * An HTML element type or a custom React element type to render as.
     */
    as?: ElementType;
    /**
     * [Slot](?path=/docs/getting-started-slots--page) to render into.
     */
    slot?: string;
    /**
     * React children
     */
    children?: ReactNode;
    /**
    * @ignore
    */
    forwardedRef: ForwardedRef<any>;
}

....

export const Text = slot("text", forwardRef<any, Omit<InnerTextProps, "forwardedRef">((props, ref) => (
    <InnerText {...props} forwardedRef={ref} />
)));
```

Reply:
There has been a couple iteration, but what i found the most effective is this : 

```
export interface InnerTextProps {
    /**
     * A text can vary in size.
     */
    size?: "2xs" | "xs" | "sm" | "md" | "lg" | "xl" | "2xl" | "3xl" | "4xl" | "5xl" | "inherit";
    /**
     * An HTML element type or a custom React element type to render as.
     */
    as?: ElementType;
    /**
     * [Slot](?path=/docs/getting-started-slots--page) to render into.
     */
    slot?: string;
    /**
     * React children
     */
    children?: ReactNode;
    /**
    * @ignore
    */
    forwardedRef: ForwardedRef<any>;
}

export type TextProps = InnerPropsToProps<InnerTextProps>;



export function InnerText(props: InnerTextProps): ReactElement {
ééé
}


export const Text = slot("text", forwardRef<any, TextProps>((props, ref) => (
    <InnerText {...props} forwardedRef={ref} />
)));
```
The issue with the 2nd option you proposed, is that if you need the BoxProps somewhere else in the code, you can't access it since you never named it. 

for instance : 

```

        <Flex
            {...mergeProps<Partial<FlexProps>[]>(
                rest,
                alignProps,
                {
                    gap: gap !== 0 ? gap : undefined,
                    wrap: !isNil(wrap) ? "wrap" : undefined,
                    ref: forwardedRef
                }
            )}
        >
            {children}
        </Flex>
```

i need the FlexProps here, and i need it to have the "ref" property.